### PR TITLE
go@1.19: deprecate

### PR DESCRIPTION
**Pull request by chenrui333**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

followup of https://github.com/Homebrew/homebrew-core/pull/134468

### CODE REVIEW

**Comment by chenrui333** [comment]
will check on the dependants.

File: Formula/g/go@1.19.rb [hunk 1]
```diff
@@ -23,6 +18,10 @@ class GoAT119 < Formula
 
   keg_only :versioned_formula
 
+  # EOL with Go 1.21 release (2023-08-08)
+  # Ref: https://go.dev/doc/devel/release#policy
+  deprecate! date: "2023-08-13", because: :unsupported
```
Comment (SMillerDev) [inline Formula/g/go@1.19.rb:23]: if it's EOL, maybe just disable it in a year?

Comment (chenrui333) [reply Formula/g/go@1.19.rb:23]: you mean change it to disable?

it does have sizable traffic to it, https://formulae.brew.sh/formula/go@1.19 (I would prefer deprecate now vs disable now :) )

**Comment by chenrui333** [comment]
ok, all the go@1.19 related formulae have been either deprecated or migrated to go@1.20, this is good to go now.

**Review by ZhongRuoyu — APPROVED** [review]

**Pull request merged**